### ISSUE TITLE
Accessible Focus from Menus on Windows

### ISSUE THREAD

**Issue by baconpaul** [surge-synthesizer/surge#6426]
Conversation with x0 (@ultrasound1372 ) and @pietermach on discord today

x0:

Okay re focus out of menus thing. Whether pressing escape or enter it does leave me in limbo most of the time until I tab, at which point it brings me back to where it was. Asking NVDA for info on the focused object after doing this gives no message at all. Opening the log viewer to look at developer info for the current navigator gives this exception text for states, is focusable, UIA automation ID, and UIA className. Exception: (-2147220991, 'An event was unable to invoke any of the subscribers', (None, None, None, 0, None))
The location is said to be a rectangle of nothing, I.E. left/top coordinates are 0, and width/height are 0, has no name, role is a window, window class name is JUCE_1823b9a25ef, and control ID, style and extended style are 0. It looks like some kind of dummy object.
This is the info on the limbo object that I end up landed on, not the object after hitting tab obviously.

What's interesting is that pressing tab again brings you to the last control that was focused, not the one after it in the order like you'd expect. Which means your tracking does work. So patching on menu close to basically simulate some of the things tab does would likely be enough.

baconpaul 
yeah so I think @pitermach gets it right: The focus stays on the vanished menu on close even though the next focused object is the original slider. If a sighted user could use the UI on windows with the focus debugger and try escaping out of a menu, that might help

x0
I need to test something actually... The way NVDA acts is as if the menu pops a separate top-level window, which then disappears afterwards. It possibly doesn't get notified of that disappearance... My evidence for this is that when activating a menu it goes Surge XT again as if it were reporting a window title change.
Focus issues are common when things only visually, and not logically, overlay other things.
Indeed @pitermach is correct, when I press enter on a menu to, say, pick an oscillator type and end up in limbo where navigation commands don't work, the log viewer shows that the supposed object with focus has a role of menu item, and throws all those exceptions. I assume window is what you get when you hit escape.
Perhaps there's some kind of destroy event that Juce doesn't call within its menus? An event that says hey, this object is now gone. Find another.

**Comment by baconpaul** [comment]
pitermach says:

If the Juce devs need help debugging UIA issues like this, it's probably worth asking Leonard de Ruijter (leonardder on github). He's responsible for much of NVDA's UIA modernisation and OSARA's UIA support he knows his way around it very well

**Comment by mkruselj** [comment]
@baconpaul Going by the last sentence pitermach said above, should we forward this issue to the JUCE crew and close it here, then?

**Comment by baconpaul** [comment]
Oh no I actually have Claude writing me a patch which I haven’t had time to review

**Comment by baconpaul** [comment]
Oh sorry I thought this was the open on move cursor one … ignore that comment

**Comment by mkruselj** [comment]
So, good to close then? 🙂
